### DISCUSSION TITLE
Feature Requests for BMW iX

### DISCUSSION THREAD

**Discussion by mobilewares** [bimmerconnected/bimmer_connected#402] [Feature request]
Hi - I'm using this with our new BMW iX - thanks for the great work!

On top of the regular HA web experience I have my own custom .NET/MAUI Android tablet app which is displaying information for our car directly from the HA API's so we can see it around the house on some tablets we have mounted.

I'd love to see some of this additional information coming through from the BMW API if possible in future releases : 

- **Climate Control on/off sensor** like we get in the App. I think the ability to enable it via service command is there but no sensor comes back to say if it's on/off. If it's in the API, the current temperature in car would be awesome too (so we can decide if it needs to be switched on/off).
- **Heading direction for GPS Location.** The BMW App shows this - very useful to determine if car is on way out/back when my wife or I want to know where eachother are. Speed would be awesome too - but not sure if API reports that.
- **Car Running Status** - haven't found any accurate way to determine if car is parked or moving other than to watch for movement updates and make a guess derived on the lock status/gps location.
- **Car Charging History** - can get this in the BMW App - assuming this is probably in API as the app has it. 

thanks!
Niall

**Comment by rikroe** [comment]
Thanks for your ideas!

* **Climate Control on/off** is on the todo list, but may be a little delayed, depending on how fast I can finish #400.
* **Heading direction for GPS** should be easily possible, I'll have a look into this.
* I don't think **Car Running Status** is available from the API directly, so I'd rather not implement it in the library. This would be an approximation at best and nothing I'd like to support --> Probably better handled using HA template sensors/automations.
* We should be able to provide **Car Charging History** as well (I've already seen some of the BMW API responses), however it'll be interesting on how to best bring this into a HA sensor format.

**Reply by demel42** [reply]
- Heading
is attribute ‚properties.vehicleLocation.heading’ in the data of ‚/eadrax-vcs/v1/vehicles'

- Car Running Status:
i think, it is the attribute ‚properties.inMotion’ in the data if ‚/eadrax-vcs/v1/vehicles'

**Reply by rikroe** [reply]
We already have `heading` as an attribute in `ConnectedDriveVehicle.status`, so it's just the matter of getting it into Home Assistant.

Actually forgot about the `inMotion` property (which is available as `ConnectedDriveVehicle.status.is_vehicle_active`). Not totally sure how it works though and which cars support this.
@demel42, @mobilewars: would you be able to get a `fingerprint` if you know that somebody else is driving your car?

**Reply by demel42** [reply]
yes, will do tomorrow.
but i am sure, it‘s a bool (1=car is driving) using in my pjp-interface

**Reply by gerard33** [reply]
I have tested that a while ago with my X1 from 2017 and that didn't support it, so probably only recent cars are supported. But it's a nice feature.

**Reply by demel42** [reply]
i have problems to generate a fingerprint (opened a issue "unable to execute new update bimmerconnected #406")

**Reply by demel42** [reply]
unfortunately I could not cause a change of the attribute inMotion today. 
I was pretty sure I've seen the variable set to true before, but maybe it was from another vehicle whose debug I saw ....

It was only a short ride of a few minutes, I will check it again when the ride is longer.

It's a 2017 X3, I'll have access to an iX3 in about 2 months.

**Reply by rikroe** [reply]
It would be very interesting if we are somehow able to determine if a vehicle has live updates or not.
Maybe [`telematicsUnit`](https://github.com/bimmerconnected/bimmer_connected/blob/master/test/responses/G30/vehicles_v2_bmw_0.json#L373), [`headUnit`](https://github.com/bimmerconnected/bimmer_connected/blob/master/test/responses/G30/vehicles_v2_bmw_0.json#L105) or [`hmiVersion`](https://github.com/bimmerconnected/bimmer_connected/blob/master/test/responses/G30/vehicles_v2_bmw_0.json#L106) could be used?
